intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+#### Cred-minting philosophy
+
+As a matter of design philosophy, we believe cred should be minted at moments where there is review of an action, rather than the action itself. This reduces the danger of spam and reinforces the idea that contributors dictate what is valuable. For other projects adopting SourceCred, contributions that mint cred could also be actions performed by a moderator or maintainer.
+
+#### Summary up to this point:
+
+In this way, PageRank takes weights in and outputs scores that value contributions based on their relationship to each other. The eigenvector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> then consists of the Cred scores for the various nodes, including the contributors, so we’ve accomplished our goal!
+
+## Cred over time
+
+#### Time-based Cred constraints
+
+SourceCred has additionally innovated on PageRank to accommodate properties that Cred should demonstrate over time. The most important properties in this respect are that:
+
+-   a contributor should never lose cred. They ought to become more diluted as other contributors continue to earn new cred, but they’ll still continue to earn Grain on that Cred.
+-   a past action that was recently found to be valuable in the present should cause new Cred to flow to that action.
+
+#### Time epochs and fibration
+
+To simplify the computational overhead and storage required to compute Cred over time, SourceCred is moving toward a fibration model where a node is added per contributor, per unit time (currently weekly for SourceCred). In this model, epoch nodes intermediate flows to a user from any nodes during a period of time, and have a fixed transition probability to the user’s node, rather than a weight from which transition probabilities are calculated. The remaining transition probability is then shared between the seed node (for the teleportation vector) and the other node weights. This will be implemented by distinguishing graph edges, which have fixed weights, from markov edges, which have fixed transition probabilities from epoch nodes. This is then input into the mechanism that calculates the transition probabilities.
+
+## Grain dynamics
+
+#### How Grain is generated
+
+Once SourceCred calculates the Cred for individual contributors, that Cred is linked to an identity that is an amalgamation of their identities across the platforms (e.g. GitHub handle or Discord username). A constant amount of SourceCred Grain (¤) is produced per unit time, currently 15000¤ per week. This amount is distributed according to two different distribution mechanisms.
+
+-   20% of the grain is “fast,” and is disbursed to nodes in proportion to the cred earned by each node in the past week. This is intended to reward new participants.
+-   80% of the grain is “slow,” and is disbursed in proportion to the difference between total grain you've been distributed, and what that node’s proportional fair Cred would be over all history
```

Comment:
Basically, the "slow" distribution tries to ensure that everyone is rewarded fairly based on their lifetime contributions. This is particularly important for people who were under-recognized in the past; say that last year you did work that no-one thought was important, but actually was quite pivotal. Last year, you got underpaid because we didn't think it was worth much; but now that your cred is higher, we should make you whole. the slow payment will "catch you up" for your past underpayment.

On the other hand, it's possible we'll change the algorithm or weights such that we now think you were over-paid in the past. If so, you won't receive any slow payments for a while.

There are some interesting second-order dynamics here relating the rate of "cred inflation" vs "grain inflation".

- If we're increasing grain faster than cred, then the slow payment will pay all contributors (even inactive contributors) because the lifetime grain-per-cred value is increasing
- If we're inflating cred faster than grain, then the slow payment will only pay recipients of "new" cred, i.e. an inactive contributor whose cred stays constant would stop receiving slow payments (because they already reached target grain-per-cred for their amount of cred)

Reply:
This is great.  I added in some of this explanation+motivation about the slow distribution, and basically copied the second-order dynamics of cred/grain inflation - hope that's okay.